bin/compile

```diff
@@ -153,6 +153,10 @@ install_bins() {
   npm_engine=$(read_json "$BUILD_DIR/package.json" ".engines.npm")
   yarn_engine=$(read_json "$BUILD_DIR/package.json" ".engines.yarn")
 
+  if [[ "$YARN2" == "true" && -n "$yarn_engine" ]]; then
+    warn " You should be using a vendored version of Yarn. Heroku will ignore the engine from the package.json. "
```

Comment:
I might just be out of the loop on yarn 2 stuff, but would an average user understand what they would need to do here to resolve this? A link to a kb or devcenter article for how to set this up might be a helpful inclusion here

Reply:
That's a good point. I would say "yes, they should know", but that's not a great DX either way. One of the things we have to do after this is set up migration docs for Yarn 2 and also adjust the current Node.js Support docs, so one of the steps for that will probably be to do a scan of all the places in the buildpack that we should link to docs for more information.